tools/src/wyvern/target/corewyvernIL/metadata/HasMetadata.java

```diff
@@ -0,0 +1,9 @@
+package wyvern.target.corewyvernIL.metadata;
+
+import java.util.Iterator;
+
+public interface HasMetadata {
+    public Metadata[] getMetadata();
```

Comment:
Both implementations store this using a Set implementation, and Set is a higher-level interface than a raw array [].  Is there any reason not to make this return a Set<Metadata>?